packages/grid/_modules_/grid/hooks/features/focus/useGridFocus.ts

```diff
@@ -80,6 +90,48 @@ export const useGridFocus = (apiRef: GridApiRef): void => {
     }));
   }, [logger, setGridState]);
 
+  const handleCellMouseUp = React.useCallback(
+    (params: GridCellParams) => {
+      const { cell } = apiRef.current.getState().focus;
+      if (!cell) {
+        return;
+      }
+
+      if (params.id === cell.id && params.field === cell.field) {
+        insideFocusedCell.current = true;
+      }
+    },
+    [apiRef],
+  );
+
+  const handleDocumentClick = React.useCallback(
+    (event) => {
+      const isInsideFocusedCell = insideFocusedCell.current;
+      insideFocusedCell.current = false;
+
+      const { cell } = apiRef.current.getState().focus;
+      if (!cell || isInsideFocusedCell) {
+        return;
+      }
+
```

Comment:
Will this work?

```suggestion
      const { cell } = apiRef.current.getState().focus;
      if (!cell || insideFocusedCell.current) {
        return;
      }
      
      insideFocusedCell.current = false;
```

Reply:
There's a problem with setting `insideFocusedCell.current` to false after checking if there's a cell focused. If there's no cell focused it will stay true, which can prevent the next click from working properly. I borrowed this logic from ClickAwayListener:

https://github.com/mui-org/material-ui/blob/aafb54acb7de52299f4c8cb14f0d4adcaac5cb90/packages/material-ui/src/ClickAwayListener/ClickAwayListener.tsx#L104-L105